addon/modifiers/au-floating-ui.js

```diff
@@ -0,0 +1,161 @@
+      referenceElement instanceof HTMLElement
+    );
+
+    assert(
+      `AuFloatingUI (modifier): No floating element was defined.`,
+      floatingElement instanceof HTMLElement
+    );
+
+    assert(
+      `AuFloatingUI (modifier): The reference and floating elements cannot be the same element.`,
+      floatingElement !== _referenceElement
+    );
+
+    assert(
+      `AuFloatingUI (modifier): @placement must start with either 'bottom' or 'top'.`,
+      defaultPlacement.startsWith('bottom') ||
+        defaultPlacement.startsWith('top')
+    );
+
+    Object.assign(floatingElement.style, {
+      position: 'fixed',
+      top: '0',
+      left: '0',
+    });
+
+    let middleware = [
+      offset(options.floater.offset),
+      flip(),
+      hide({ strategy: 'referenceHidden' }),
+      hide({ strategy: 'escaped' }),
+    ];
+
+    if (arrowElement) {
+      middleware.push(
+        arrow({
+          element: arrowElement,
+          padding: options.arrow.padding,
+        })
+      );
+    }
+
+    let update = async () => {
+      let { x, y, placement, middlewareData } = await computePosition(
+        referenceElement,
+        floatingElement,
+        {
+          middleware,
+          placement: defaultPlacement,
+        }
+      );
+
+      Object.assign(floatingElement.style, {
+        transform: `translate3d(${Math.round(x)}px, ${Math.round(y)}px, 0)`,
+        visibility: middlewareData.hide?.referenceHidden ? 'hidden' : 'visible',
+      });
+
+      if (middlewareData.arrow) {
+        const { x } = middlewareData.arrow;
+        const [side, alignment] = placement.split('-');
+        const isAligned = alignment != null;
+
+        const unsetSides = {
+          top: '',
+          bottom: '',
+          left: '',
+          right: '',
+        };
+
+        const mainSide = {
+          top: 'bottom',
+          bottom: 'top',
+        }[side];
+
+        const rotation = {
+          top: '180deg',
+          bottom: '0deg',
+        }[side];
+
+        const crossSide = {
+          'top-start': 'left',
+          'top-end': 'right',
+          'bottom-start': 'left',
+          'bottom-end': 'right',
+        }[placement];
+
+        Object.assign(arrowElement.style, {
+          ...unsetSides,
+          transform: `rotate(${rotation})`,
+        });
+
+        if (isAligned) {
+          Object.assign(arrowElement.style, {
+            [crossSide]:
+              typeof options.arrow.position === 'string'
+                ? options.arrow.position
+                : `${options.arrow.position}px`,
+          });
+        } else {
+          Object.assign(arrowElement.style, {
+            left: x != null ? `${x}px` : '',
+          });
+        }
+
+        Object.assign(arrowElement.style, {
+          [mainSide]: `${-options.arrow.offset}px`,
+        });
+      }
+    };
+
+    let cleanup = autoUpdate(referenceElement, floatingElement, update);
+
+    return () => {
+      cleanup();
```

Comment:
@Windvis I noticed something weird when testing the cleanup function: when utilising the floating-ui abilities as a modifier, the cleanup (destroyer) function gets called as you would expect. But when I use the `AuFloatingUi` component this doesn't happen for a reason (maybe it doesn't have to?). Do you have any experience regarding modifiers? 🤞

Reply:
I would expect that the component version works the same. How did you test it exactly? Maybe it's a bug in the `modifier` helper?

Reply:
I was thinking in the same direction, so I made a bug report within `ember-modifier` at https://github.com/ember-modifier/ember-modifier/issues/613 the previous week. Fingers crossed it is related to that helper 🤞

Reply:
Since this is for internal use only, can't we simply drop the component version for now and use the modifier directly? That would unblock your dropdown issue at least I think?

Reply:
@Windvis I'll give this a go tomorrow and let you know if this a possible solution 👍

Reply:
I gave it a go within #349 (as I have an ideal test case already there). Seems to work fine just as a modifier.

Does these changes look ok for you? https://github.com/appuniversum/ember-appuniversum/pull/349/commits/fd16e8b89a3869bd114051ffed274e881c75291b

Reply:
Left some more notes, other than that I think it's good to go if everything works as expected.